# ChemSpider

Experimental logP records of compounds in drug-like set was checked by manually querying Chemspider website. Experimental logP values (if exists), were under `Properties/Experimental Physico-chemical Properties/Experimental LogP`. I recorded these values under Chemspider columb in `df_drug_final_chemspider_logP.csv`.

In [1]:
import pandas as pd
import pickle

In [2]:
df_chemspider = pd.read_csv("df_drug_final_chemspider_logP.csv")
df_chemspider

eMolecules ID                          canonical isomeric SMILES  \
0        2010194       c1ccc(cc1)c2c3c(ncnc3oc2c4ccccc4)NCCN5CCOCC5   
1        4375254  CCOC(=O)c1ccc(cc1)Nc2cc(nc(n2)Nc3ccc(cc3)C(=O)...   
2       18897105  c1ccc2c(c1)c(=O)[nH]c(n2)CCC(=O)Nc3ncc(s3)Cc4c...   
3        1574612          c1cc(cc(c1)Br)Nc2c(cnc(n2)Nc3cccc(c3)Br)F   
4        3365457        CCOc1ccc2c(c1)sc(n2)NC(=O)Cc3ccc(c(c3)Cl)Cl   
5       46568819   c1cc(cc(c1)OCc2ccc(cc2Cl)Cl)/C=C/3\C(=O)NC(=O)S3   
6        3064762   c1ccc(cc1)C(=O)Nc2ccc(cc2)Oc3c4c5c(sc4ncn3)CCCC5   
7       10794751   CC(C)(C)c1cc(n(n1)c2ccccc2)NC(=O)Nc3cccc(c3Cl)Cl   
8        5428718        c1ccc(c(c1)C(=O)Nc2nnc(s2)SCc3ccc(cc3)Br)Cl   
9        1415746        COc1ccc(cc1)c2c3c(ncnc3oc2c4ccc(cc4)OC)NCCO   

                                   eMolecules SMILES   pKas in [3,11]  XlogP  \
0     O1CCN(CC1)CCNc1ncnc2c1c(c1ccccc1)c(o2)c1ccccc1   [3.523, 7.492]  3.254   
1  CCOC(=O)c1ccc(cc1)Nc1cc(C)nc(n1)Nc1ccc(cc1)C(=...          [6.336]  2.937   
2  O=C(Nc1ncc(s1)Cc1ccc(c(c1)F)F)CCc1nc2ccccc2c(=...  [9.381, 10.773]  3.341   
3            Brc1cccc(c1)Nc1ncc(c(n1)Nc1cccc(c1)Br)F          [3.892]  4.140   
4        CCOc1ccc2c(c1)sc(n2)NC(=O)Cc1ccc(c(c1)Cl)Cl          [9.167]  5.171   
5     O=C1NC(=O)/C(=C\c2cccc(c2)OCc2ccc(cc2Cl)Cl)/S1           [8.05]  5.238   
6       O=C(c1ccccc1)Nc1ccc(cc1)Oc1ncnc2c1c1CCCCc1s2          [3.199]  4.718   
7     O=C(Nc1cccc(c1Cl)Cl)Nc1cc(nn1c1ccccc1)C(C)(C)C          [4.113]  5.784   
8            Brc1ccc(cc1)CSc1nnc(s1)NC(=O)c1ccccc1Cl          [6.525]  5.900   
9      OCCNc1ncnc2c1c(c1ccc(cc1)OC)c(o2)c1ccc(cc1)OC          [4.829]  2.794   

     MolWt  Availability (mg)  Price      group  N_Rot  N_UV_chrom Selection  \
0  400.473              110.0    168  drug-like      6          71    picked   
1  420.461              319.0    168  drug-like     10          28    picked   
2  426.439              247.7    223  drug-like      7          37    picked   
3  438.092              222.0    168  drug-like      4          28    picked   
4  381.276              489.9    148  drug-like      6          28    picked   
5  380.245              154.0    219  drug-like      4          24    picked   
6  401.481              636.9    249  drug-like      5          34    picked   
7  403.305              324.5    148  drug-like      6          35    picked   
8  440.765              239.5    400  drug-like      6          24    picked   
9  391.420              398.0    168  drug-like      7          71    picked   

   Bin index  Priority  Final list               Chemspider  
0        0.0         2        True  3.643 Vitas-M STK198606  
1        0.0         3        True                     None  
2        0.0         1        True                     None  
3        2.0         1        True                     None  
4        3.0         1        True                     None  
5        3.0         2        True                     None  
6        3.0         3        True                     None  
7        4.0         1        True                     None  
8        4.0         2        True                     None  
9        NaN         1        True                     None

In [3]:
df_chemspider_exp_logP = df_chemspider.loc[df_chemspider["Chemspider"] != "None"]
print("{} chemicals have experimental logP reprted in ChemSpider.".format(df_chemspider_exp_logP.shape[0]))

1 chemicals have experimental logP reprted in ChemSpider.


I will save lists of chemicals selected for fragment-like set that have reported experimental logP in ChemSpider in lists (pickle files) for each round. eMolecules IDs below are listed here because they are detected to have experimental logP available.

#### Round 1 
[4424334, 17664146]

#### Round 2
[2010194]

#### Round 3

#### Round 4


In [4]:
drugs_with_exp_logP = list(df_chemspider_exp_logP.loc[:,"eMolecules ID"])
filename = "drugs_with_exp_logP_round2.pickle"
pickle.dump(drugs_with_exp_logP, open(filename, "wb"))

In [5]:
drugs_with_exp_logP_list = pickle.load(open(filename, "rb"))
print(drugs_with_exp_logP_list)

[2010194]
